data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/DataType.java

```diff
@@ -49,6 +50,13 @@ public enum DataType {
      */
     DOUBLE("double"),
 
+    /**
+     * Type of <i>BigDecimal</i>. No precision loss possible type. Compatible with the Java <b>BigDecimal</b> primitive data type.
+     *
+     * @since 2.8
+     */
+    BIGDECIMAL("bigdecimal"),
```

Comment:
Should we just call this `decimal`?

If we keep the big, we should name the enum to `BIG_DECIMAL`. I also think that we should have an underscore in the type name to keep consistency: `big_decimal`. This would match field names in OpenSearch like `geo_point`, etc.

Reply:
Kept the word `BIG` and renamed the references to `BIG_DECIMAL`. Both enum and data type name are now modified.